common/components/common/upload/ImageCropUploadButton.jsx

```diff
@@ -0,0 +1,197 @@
+// @flow
+
+import React from 'react';
+import {Button} from 'react-bootstrap';
+import type { S3Data } from './S3Data.jsx'
+import FileSelectButton from './FileSelectButton.jsx';
+import ReactCrop from "react-image-crop";
+
+export type FileUploadData = {|
+  key: string,
+  fileName: string,
+  publicUrl: string
+|};
+
+type Props = {|
+  onFileUpload: (FileUploadData) => void,
+  buttonText: string,
+  cropButtonText: string,
+  acceptedFileTypes: string,
+  iconClass: string
+|};
+
+type State = {|
+  s3Key: string,
+|};
+
+class ImageCropUploadButton extends React.PureComponent<Props, State> {
+
+  constructor(): void {
+    super();
+    this.state = {
+      s3Key: "",
+      src: null,
+      isCropping: false,
+      crop: {
+        unit: "%",
+        width: 30,
+        aspect: 16 / 9
```

Comment:
Aspect should be a prop, as not all images will have the same ratio.  The Project icon should be 1:1, for instance.